src/extension/prelude.rs

```diff
@@ -82,6 +100,21 @@ pub fn new_array(typ: Type, size: u64) -> Type {
     Type::new_extension(custom_t)
 }
 
+/// Name of the operation in the prelude for creating new arrays.
+pub const NEW_ARRAY_OP_ID: &str = "new_array";
+
+/// Initialize a new array op of type `typ` of length `size`
```

Comment:
super-nit: we talk about the "type" of an array, but the array is itself an array; the thing we are referring to is the *element type*.

(You could rename `typ` to `elem_ty` or similar if you wanted, but if not, then super-super-nit `typ` -> `ty` would make the subsequent construction easier)